feat: Add tabs to display Read and Write methods in Debug Contracts page

wugalde19 opened the pull request:
# Separate Read and Write Tabs in Debug Contracts

Fixes https://github.com/Scaffold-Stark/scaffold-stark-2/issues/281

## Short description of the solution
Replaced the separate `Read` and `Write` sections with a single tabbed interface using [DaisyUI's](https://daisyui.com/components/tab/) tabs.

Also, as suggested in the issue made use of [next/dynamic](https://nextjs.org/docs/app/building-your-application/optimizing/lazy-loading) to lazy load the content of the `Write` tab

## Types of change

- [ ] Feature
- [ ] Bug
- [X] Enhancement

## Comments (optional)
NOTE for reviewers: As mentioned in the telegram channel there are many styles for the tabs. On this PR I added 4 different commits with 4 different proposals. Here is what each of them looks like, please let me know which one you like the most to either remove the other commits or just apply the change of the option you like the most. And also, if you have another proposal please let me know

### Option 1
<img width="1036" alt="Screenshot 2024-09-28 at 2 30 04 PM" src="https://github.com/user-attachments/assets/671790e1-d41e-4e22-a3b9-20e37135fcd5">

### Option 2
<img width="1037" alt="Screenshot 2024-09-28 at 2 30 24 PM" src="https://github.com/user-attachments/assets/559ad8e0-3ef6-4af8-8d36-a5ced1396540">

### Option 3
<img width="1035" alt="Screenshot 2024-09-28 at 2 29 04 PM" src="https://github.com/user-attachments/assets/371d8e1a-02bf-4d44-baa3-3213b01eb652">

### Option 4
<img width="1047" alt="Screenshot 2024-09-28 at 2 26 36 PM" src="https://github.com/user-attachments/assets/0bae83ea-2cfd-4474-a558-d6fd1a0a386c">

### How it was before (for reference)
<img width="1064" alt="Screenshot 2024-09-28 at 1 48 12 PM" src="https://github.com/user-attachments/assets/01907e3a-8334-42c6-8430-221cfd0e86df">

metalboyrick commented on packages/nextjs/app/debug/_components/contract/ContractUI.tsx:
```diff
@@ -97,7 +97,7 @@ export const ContractUI = ({
 
         <div className="col-span-1 lg:col-span-2 flex flex-col gap-6">
-          <div className="tabs tabs-boxed">
+          <div className="tabs tabs-boxed border border-[#8A45FC]">
```
Hey @wugalde19 , thanks alot for the PR! Really appreciate it

I personally prefer this design, but with less border radius, just to keep it consistent with the other boxes!

metalboyrick replied:
by "this design" i meant option 2

Nadai2010 replied:
I like 2

wugalde19 replied:
@Nadai2010 @metalboyrick Great
By less border radius you mean something like this: 
<img width="1028" alt="image" src="https://github.com/user-attachments/assets/5d60d4ed-4b57-4476-9c26-0ee77cfe3964">

Note: I'm using the same 5px (for both the tabs container and the active tab) that the other boxes use

Nadai2010 replied:
I just happened to upload this change, xd very good, you can check the tone of the background a little, in the light you see a light gray/blue tone, and in the dark a dark gray, try to make it follow the line with the rest, (maybe transparent)

wugalde19 replied:
Transparent sounds like a good idea
Just push the latest changes @Nadai2010 
Wanna take a look?

Nadai2010 replied:
I'm checking now

Nadai2010 reviewed (changes requested):
Hi @wugalde19, thanks for showing these different options! We are sticking with this version 2 of the [commit](https://github.com/Scaffold-Stark/scaffold-stark-2/pull/297/commits/21f0ee484e3af82df20e5106db5e9300e4874026), but as @metalboyrick mentioned, with less border radius, just to keep it consistent with the other boxes. Also, please check the background color in different light and dark tones to ensure it is as consistent as possible with the rest of the boxes!

![image](https://github.com/user-attachments/assets/5b24b10b-2144-4aa1-ba62-4988a0969a3f)

![image](https://github.com/user-attachments/assets/4a29a6ca-7655-4e0a-8505-4fd03097b6e1)

wugalde19 commented:
@Nadai2010 @metalboyrick 
Here's what the latest version looks like after addressing all your comments

### Dark mode
<img width="1024" alt="image" src="https://github.com/user-attachments/assets/2afc4e35-c5de-4198-b4c1-0691df1d353a">

### Light mode
<img width="1019" alt="image" src="https://github.com/user-attachments/assets/02c7d107-f016-4a96-9033-b2bf0a6c383e">

metalboyrick reviewed (approved):
Looking good, of the same opinion as @Nadai2010, thanks for the hard work!

wugalde19 commented:
@metalboyrick @Nadai2010 Anytime!
Thanks for all your valuable feedback
